### PR TITLE
MueLu: Small updates on ReorderBlockAFactory

### PR DESCRIPTION
**Pull request by maxfirmbach**
@trilinos/muelu

## Motivation

If the reordering string is empty we run into problems e.g. if I have cases where I just want to split a matrix and don't want a reordering at all. This might happen in fluid problems splitting the matrix into velocity and pressure parts.

On the way I did some minor cleanup.

Not sure if in the future it would make sense to separate functionalities of splitting and reordering (in some sense splitting is also some kind of reordering :smile:).

## Stakeholder Feedback

Related to physics-based block-preconditioning.

## Additional Information

Also interesting for @mayrmt @Max-pert

### CODE REVIEW

**Comment by GrahamBenHarper** [comment]
The AT says it failed, but the results don't show any failures... maybe they're incomplete like @jhux2 recently experienced. Can you confirm @jhux2?

File: packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_def.hpp [hunk 1]
```diff
@@ -143,27 +143,39 @@ namespace MueLu {
...
-    Teuchos::RCP<ReorderedBlockedCrsMatrix> bret =
-        Teuchos::rcp_const_cast<ReorderedBlockedCrsMatrix>(brop);
-
-    // strided maps
-    // blocked operators do not have strided maps (information could be misleading?)
-    //Op->CreateView("stridedMaps", srangeMap, sdomainMap);
+    // if we have a blocked operator and a reordering string, create a nested blocked operator, if not skip the process
+    if(reorderStr.empty())
+    {
+      GetOStream(Statistics1) << "Skipping reordering A" << std::endl;
```
Comment (mayrmt) [inline packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_def.hpp:149]:
```suggestion
      GetOStream(Statistics1) << "No reordering information provided. Skipping reordering of A." << std::endl;
```

File: packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_decl.hpp [hunk 2]
```diff
@@ -84,10 +84,8 @@ namespace MueLu {
     //@{
 
     //! Constructor.
-    ReorderBlockAFactory() { }
+    ReorderBlockAFactory() = default;
```
Comment (mayrmt) [inline packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_decl.hpp:87]: You can even delete this one as well. Then, the compiler will generate the default constructor by itself.

File: packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_def.hpp [hunk 3]
```diff
@@ -143,27 +143,39 @@ namespace MueLu {
...
+      Teuchos::RCP<const ReorderedBlockedCrsMatrix> brop =
+              Teuchos::rcp_dynamic_cast<const ReorderedBlockedCrsMatrix>(
+                      Xpetra::buildReorderedBlockedCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>(brm, A));
+
+      TEUCHOS_TEST_FOR_EXCEPTION(brop.is_null(), Exceptions::RuntimeError,
+                                 "Block reordering of " << A->Rows() << "x" << A->Cols()
+                                                        << " blocked operator failed.");
+
+      GetOStream(Statistics1) << "Reordering A using " << brm->toString() << " block gives a " << brop->Rows() << "x"
+                              << brop->Cols() << " blocked operators" << std::endl;
```
Comment (mayrmt) [inline packages/muelu/src/Transfers/BlockedTransfers/MueLu_ReorderBlockAFactory_def.hpp:165]: Typo:
```suggestion
                              << brop->Cols() << " blocked operator" << std::endl;
```

**Review by mayrmt — COMMENTED** [review]
Just some minor remarks

**Comment by GrahamBenHarper** [comment]
@sebrowne are you able to see why these tests failed? The AT reports the tests failed, but all the tests that ran 20 hours ago report no failures on the CDash site. I'm worried this may be a repeat of issue #11956.

**Comment by sebrowne** [comment]
It failed because CDash was down temporarily (it appears), so the submission of the (passed) build/test XML failed, which then reported a failure to the autotester.  Definitely an AT:RETEST case

**Comment by GrahamBenHarper** [comment]
Thank you @sebrowne!

**Comment by maxfirmbach** [comment]
Seems like the AT passes now :+1:. I did some minor changes, so this would be ready for another round of testing.

**Review by mayrmt — APPROVED** [review]

**Comment by mayrmt** [comment]
@maxfirmbach Testing should be on its way. Is this ready to be merged from your point of view? Then, I'll set the respective label.

**Comment by maxfirmbach** [comment]
> @maxfirmbach Testing should be on its way. Is this ready to be merged from your point of view? Then, I'll set the respective label.

Yes, this is ready to be merged.

**Pull request merged**